### ISSUE TITLE
Keyboard navigation between data-grid header and rows should be done with the arrow keys not the tab

### ISSUE THREAD

**Issue by limonte** [vaadin/vaadin-grid#1191]
### Steps to reproduce
1. Tab to the vaadin-grid header
2. Press the down arrow key

**Expected behavior:** Focus moves to the first item in the first row of the vaadin-grid
**Actual behavior:** Nothing happens

### Description
The user shouldn’t have to tab between the column headers and the column data, that should be navigated using the arrow keys, as in the Example 2 here: https://www.w3.org/TR/wai-aria-practices/examples/grid/dataGrids.html

**Comment by tomivirkki** [comment]
What if you have a million rows in the grid? Does the user have to press the down arrow key a million times to focus the footer items?

**Comment by limonte** [comment]
> What if you have a million rows in the grid? Does the user have to press the down arrow key a million times to focus the footer items?

<kbd>Ctrl</kdb> + <kbd>End</kdb>

**Comment by tomivirkki** [comment]
That's requiring the user to leave the viewport they're working on just in order to access some controls in header/footer.

**Comment by limonte** [comment]
> That's requiring the user to leave the viewport they're working on just in order to access some controls in header/footer.

This is a valid argument. Thanks @tomivirkki.

**Comment by jouni** [comment]
The issue is still valid if the user is currently focusing the first/last row in the viewport. In those cases, up/down arrow should move the focus automatically.

And the same for the last/first header/footer row: arrow keys could move the focus to the first/last body row (which is currently visible in the viewport).

Grid v1 works this way, actually: https://demo.vaadin.com/sampler/#ui/grids-and-trees/grid/features
